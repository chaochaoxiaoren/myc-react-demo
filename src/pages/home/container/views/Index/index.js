import React from 'react';
import { Link, Outlet } from 'react-router-dom';

const Index = () => {

  return (
    <>
      <Link to="/home/market">market</Link> <br />
      <Link to="/home/doc">Doc</Link> <br />
      <Link to="/home/demo">Demo</Link> <br />
      <Link to="/home/less">Less</Link> <br />
      <Link to="/home/sass">Sass</Link> <br />
      <Link to="/home/web3">Web3</Link> <br />
      ========================================
      <Outlet />
    </>
  );
};

export default Index;
