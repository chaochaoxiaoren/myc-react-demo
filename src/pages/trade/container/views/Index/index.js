import React, { useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux'
import { decrement, increment, incrementByAmount } from '@trade/store/reducer/counter'
import style from './index.module.css';
import Loading from '@trade/components/Loading/index.jsx';
import { Link } from 'react-router-dom';

const Index = () => {
  const inputEle = useRef(null);

  const count = useSelector(state => state.counter.value)
  const dispatch = useDispatch()

  return (
    <>
      <Link to="/trade/vlist">Vlist</Link> <br />
      <Link to="/trade/intro">intro</Link> <br />
      <Link to="/trade/mock">mock</Link> <br />
      <Link to="/trade/lottie">lottie</Link> <br />
      <div className={style.body}>
        组件样式
      </div>
      <Loading />
      {count}
      <br />
      <button
        onClick={() => {
          dispatch(increment());
        }}
      >
        加一
      </button>
      <button onClick={() => dispatch(decrement())}>减一</button>
      <br />
      <input type='number' ref={inputEle} />
      <button
        onClick={() => {
          const num = Number(inputEle.current.value)
          num && dispatch(incrementByAmount(num))
        }}
      >
        加
      </button>
    </>
  );
};

export default Index;
