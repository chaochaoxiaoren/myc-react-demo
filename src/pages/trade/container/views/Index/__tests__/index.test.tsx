import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BrowserRouter } from 'react-router-dom';
import Index from "../index";

describe("Index", () => {
  test("抓取快照", () => {
    // 创建模拟的 store
    const mockStore = configureStore();
    const initialState = {counter: {
      value: 1
    } }; // 根据你的需求设置初始状态
    const store = mockStore(initialState);

    // 使用 Provider 包裹你的组件
    const { baseElement } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Index />
        </BrowserRouter>
      </Provider>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
