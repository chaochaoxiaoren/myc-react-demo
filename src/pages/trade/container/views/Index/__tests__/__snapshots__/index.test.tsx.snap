// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index 抓取快照 1`] = `
<body>
  <div>
    <a
      href="/trade/vlist"
    >
      Vlist
    </a>
     
    <br />
    <a
      href="/trade/intro"
    >
      intro
    </a>
     
    <br />
    <a
      href="/trade/mock"
    >
      mock
    </a>
     
    <br />
    <a
      href="/trade/lottie"
    >
      lottie
    </a>
     
    <br />
    <div>
      组件样式
    </div>
    <div>
      loading
    </div>
    1
    <br />
    <button>
      加一
    </button>
    <button>
      减一
    </button>
    <br />
    <input
      type="number"
    />
    <button>
      加
    </button>
  </div>
</body>
`;
